MIGRATION_GUIDE.md

```diff
@@ -1,9 +1,170 @@
+    "${snowflake_role.b.name}" = "\"${snowflake_role.b.name}\"|false|false|USAGE|OnAccountObject|DATABASE|\"${snowflake_database.test.name}\""
+  }
+}
+
+resource "snowflake_grant_privileges_to_account_role" "new_resource" {
+  depends_on = [snowflake_database.test, snowflake_role.a, snowflake_role.b]
+  for_each   = local.migrations
+  privileges = ["USAGE"]
+  account_role_name  = "\"${each.key}\""
+  on_account_object {
+    object_type = "DATABASE"
+    object_name = "\"${snowflake_database.test.name}\""
+  }
+}
+
+import {
+  to = snowflake_grant_privileges_to_account_role.new_resource[each.key]
+  id = each.value
+  for_each = local.migrations
+}
+```
+
+[Hashicorp documentation reference on import block](https://developer.hashicorp.com/terraform/language/import)
+
+#### 3.2.2 Run terraform plan and apply
+
+After running `terraform plan` you'll see if resources can be imported without any change. If that's the case
+and nothing has to be adjusted, then we can perform `terraform apply` to import the state into our new grant resources.
+
+#### 3.3.1. Write import block
+
+Unfortunately, `for_each` cannot be used when generating with import blocks, so we have to define them one by one.
+For simplicity, we'll define just one import block (the second one would look the same, only with a different role).
+
+```terraform
+import {
+  to = snowflake_grant_privileges_to_account_role.new_resource_role_a
+  id = "\"${snowflake_role.a.name}\"|false|false|USAGE|OnAccountObject|DATABASE|\"${snowflake_database.test.name}\""
+}
+```
+[Hashicorp documentation reference on import block](https://developer.hashicorp.com/terraform/language/import)
+
+#### 3.3.2. terraform plan -generate-config-out
+
+After specifying the import block run the `terraform plan -generate-config-out=generated.tf` command,
+which will scan your configuration files search for import blocks, and put the generated configurations inside the `generated.tf` file.
+
+```terraform
+# __generated__ by Terraform
+# Please review these resources and move them into your main configuration files.
+
+# __generated__ by Terraform
+resource "snowflake_grant_privileges_to_account_role" "new_resource_role_a" {
+  account_role_name    = "\"test_role_321123123\""
+  all_privileges       = false
+  always_apply         = false
+  always_apply_trigger = null
+  on_account           = false
+  privileges           = ["USAGE"]
+  with_grant_option    = false
+  on_account_object {
+    object_name = "\"test_database_1231321\""
+    object_type = "DATABASE"
+  }
+}
+```
+
+#### 3.3.3. terraform plan and apply
+
+After running `terraform plan` you'll see if there are any changes we have to do before applying our generated configuration.
+If no errors are appearing you can run `terraform apply` to import state into generated configurations. 
+
+#### 3.3.4. Thoughts on configuration generation
```

Comment:
would recommend changing the title here to something more clear. like "Limitations of Configuration Generation".